crates/telemetry/src/logs.rs

```diff
@@ -106,7 +101,9 @@ pub(crate) fn init_otel_logging_backend(spin_version: String) -> anyhow::Result<
         )
         .build();
 
-    global::set_logger_provider(provider);
+    LOGGER
+        .set(provider.logger("spin"))
+        .expect("should set logger");
```

Comment:
Do we really want to panic here?

Reply:
My assumption is that this should never fail because this is the only place we're setting the logger. I also assume that if this does fail then we don't want to continue b/c something weird is going on and we're not going to have an OTel logger.

What we do here affects how we handle the other place you left a comment.

Personally unless there is a clear reason setting a `OnceLock` would fail, I'd prefer to panic here.

Reply:
It panics if it's called more than once not only if it's called concurrently. Or are we checking just before this that the value isn't already set?

Reply:
> It panics if it's called more than once not only if it's called concurrently.

Oh sure enough; I misread the docs.

Reply:
Given :point_up: I'd suggest just ignoring a failure here rather than panicking :shrug:

Reply:
Fixed